services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```
@@ -1033,20 +1034,38 @@ public void deleteStore(
       sendAdminMessageAndWaitForConsumed(clusterName, storeName, message);
 
       // Deleting ACL needs to be the last step in store deletion process.
-      if (store != null) {
-        if (!store.isMigrating()) {
-          cleanUpAclsForStore(storeName, VeniceSystemStoreType.getEnabledSystemStoreTypes(store));
-        } else {
-          LOGGER.info("Store: {} is migrating! Skipping acl deletion!", storeName);
-        }
-      } else {
-        LOGGER.warn("Store object for {} is missing! Skipping acl deletion!", storeName);
-      }
+      deleteAclsForStore(store, storeName);
+    } catch (AdminMessageConsumptionTimeoutException timeoutException) {
+      LOGGER.info(
+          "Timed out while waiting for delete store admin message to be consumed for store: {} in cluster: {}",
+          storeName,
+          clusterName,
+          timeoutException);
+      deleteAclsForStore(store, storeName);
+    } catch (Exception e) {
+      LOGGER.info("Caught an exception when deleting store {} in cluster {}", storeName, clusterName, e);
```

So the LHS code propagate the Exception to the caller, but RHS does not, though it logs it which is a good thing.
Do we want the caller not receive the exception??

---

What are you referring to, by LHS and RHS? I don't see any equals sign here?

---

In both catch exception branches, we should throw the exception and bubble it up to the callers too.

---

Thanks for the catch - updated the change to throw the exception so it gets bubbled up to the callers too

---

but @huangminchn  If AdminMessageConsumptionTimeoutException is propagated to the caller, what action caller is supposed to take?
Also if we propagate it, it will call `veniceResponse.setError(e)` in StoresRoute and that admin operation will be marked as failed, but we do not want that because we have already taken action on this timeout exception by catching it and calling `deleteAclsForStore`, do we?